service/lib/dinstaller/dbus/storage/manager.rb

```diff
@@ -131,6 +123,75 @@ def calculate_proposal(dbus_settings)
           success ? 0 : 1
         end
 
+        dbus_interface PROPOSAL_CALCULATOR_INTERFACE do
+          dbus_reader :available_devices, "a(ssa{sv})"
+
+          dbus_reader :volume_templates, "aa{sv}"
+
+          dbus_reader :result, "o"
+
+          # result: 0 success; 1 error
+          dbus_method :Calculate, "in settings:a{sv}, out result:u" do |settings|
+            busy_while { calculate_proposal(settings) }
+          end
+        end
+
+        ISCSI_INITIATOR_INTERFACE = "org.opensuse.DInstaller.Storage1.ISCSI.Initiator"
+        private_constant :ISCSI_INITIATOR_INTERFACE
+
+        # Gets the iSCSI initiator name
+        #
+        # @return [String]
+        def initiator_name
+          backend.iscsi.initiator.name || ""
+        end
+
+        # Sets the iSCSI initiator name
+        #
+        # @param value [String]
+        def initiator_name=(value)
+          backend.iscsi.initiator.name = value
+        end
+
+        # Performs an iSCSI discovery
+        #
+        # @param address [String] IP address of the iSCSI server
+        # @param port [Integer] Port of the iSCSI server
+        # @param options [Hash<String, String>] Options from a D-Bus call:
+        #   @option InitiatorUsername [String]
+        #   @option InitiatorPassword [String]
+        #   @option TargetUsername [String]
+        #   @option TargetPassword [String]
+        #
+        # @return [Integer] 0 on success, 1 on failure
+        def iscsi_discover(address, port, options)
+          success = backend.iscsi.discover_send_targets(address, port, iscsi_auth(options))
+          success ? 0 : 1
+        end
+
+        # Deletes an iSCSI node from the database
+        #
+        # @param path [::DBus::ObjectPath]
+        # @return [Integer] 0 on success, 1 on failure
+        def iscsi_delete(path)
```

Comment:
This returns 1 both for a not found node and for a failure in deleting. And at first sight I don't see any trace in the log making any difference. I said "at first sight". Maybe the log traces left by `iscsi_nodes_tree` and `backend.iscsi.delete` are enough to spot the difference, but...